SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -968,7 +974,18 @@ void ActorSystem::ApplyComponentData(USpatialActorChannel& Channel, UObject& Tar
 		bool bOutReferencesChanged = false;
 
 		FObjectRepNotifies& ObjectRepNotifiesOut = GetObjectRepNotifies(TargetObject);
-		Reader.ApplyComponentData(ComponentId, Data, TargetObject, Channel, ObjectRepNotifiesOut, bOutReferencesChanged);
+		const bool bSuccessfullyPreNetReceived = InvokePreNetReceive(Channel, TargetObject);
+		if (bSuccessfullyPreNetReceived)
+		{
+			Reader.ApplyComponentData(ComponentId, Data, TargetObject, Channel, ObjectRepNotifiesOut, bOutReferencesChanged);
+		}
+		else
+		{
+			UE_LOG(LogActorSystem, Log,
+				   TEXT("ApplyComponentData: Did not invoke PreNetReceive for object %s, entity id %lld, component id %u. No data will be "
+						"applied."),
+				   *TargetObject.GetName(), Channel.GetEntityId(), ComponentId);
+		}
```

Comment:
might be worth noting here that with this, if an object kills itself within PreNetReceive, we no longer apply the data, whereas previously we did. (this is due to Reader.ApplyComponentData early-ing out with from an IsPendingKill check)

Doesn't seem necessarily like a bad thing to me, though could be changed

Reply:
(this also applys to ApplyComponentUpdate)

Reply:
I tried deleting an object in prenetreceive in one of my tests, and we just hard-crashed, so I think this is an improvement

Reply:
ok kool